### PR TITLE
feat(gridster): allow more columns than 6

### PR DESCRIPTION
**Pull request by raszi**

### CODE REVIEW

**Comment by raszi** [comment]
Maybe it is possible but why should we leave a bug in the dashing?

**Comment by tylermauthe** [comment]
Maybe a better solution would be to add an optional config object. Store the default config (`defaultConfig`) in an object, use a deep-copy override `overriddenConfig = jquery.extend(true, {}, defaultConfig, Dashing.gridsterOverride);` and then supply `overriddenConfig` as the parameter to gridster.

This would allow people to override any of gridster's settings on a per-dashboard basis.

I'd argue that moving into Dashing 2.0, then we should use this approach to customising all aspects of such components.

So instead of setting Dashing.widget_margins in your dashboard to customise, you'd set `Dashing.gridster.widget_margins` (or `Dashing.gridster = {widget_margins: ...}`)

**Comment by raszi** [comment]
I did not modify the widget_margins.

I just added the `max_size_x` property because that was missing from the setup therefore somebody could set the `numColumns` config in their app but the gridster still used the previous column count.

What is the reason to make `numColumns` configurable if you cannot set to any value you want?

**Comment by tylermauthe** [comment]
Let me rephrase my last sentence (which was a generalisation, not specific to you):

So instead of setting `Dashing.numColumns` in your dashboard to customise, you'd set `Dashing.gridster.max_size_x` (or `Dashing.gridster = {max_size_x: ...}`)

The reason why I make this suggestion is that _you_ may want this flag, but another developer may want to set a different flag. If we add each one by one, pretty soon we will have replicated Gridster's API and we'll have a mess of flags.

**Comment by tylermauthe** [comment]
I've already said my peace on this PR -- I don't support adding one more flag.

**Comment by pushmatrix** [comment]
I'm not able to reproduce the problem. I can set `Dashing.numColumns = 9` and I get more than six columns, and my widgets can also span the whole length. Was there a new version of gridster that put a hard limit at 6?

**Comment by DeanCording** [comment]
The problem is that widgets can only be 0-5 columns wide, irrespective of the dashboard width.  If you specify more than 5 columns wide for a widget it fills the entire dashboard width.  I encountered this issue with a dashboard I am developing.  To maximise the use of screen space, I have set the column width to 50px, with most widgets being 4 or 5 columns wide.  The problem occurs when I have a larger widget that takes up 10 columns, it fills the entire dashboard width.

I can confirm that the proposed fix solves this problem.
